test/simplify_algebra_test.cpp

```diff
@@ -804,6 +1166,147 @@ TEST_CASE(simplify_sub_const)
+        m2.add_return({zero});
+    }
+
+    EXPECT(m1 == m2);
+}
+
+TEST_CASE(simplify_zero_mult_const2)
+{
+    migraphx::module m1;
+    {
+        auto x       = m1.add_parameter("x", {migraphx::shape::int32_type, {1}});
+        auto zero    = m1.add_literal(0);
+        auto mul_ins = m1.add_instruction(migraphx::make_op("mul"), zero, x);
+        m1.add_return({mul_ins});
+    }
+    run_pass(m1);
+
+    migraphx::module m2;
+    {
+        m2.add_parameter("x", {migraphx::shape::int32_type, {1}});
+        auto zero = m2.add_literal(0);
+        m2.add_return({zero});
+    }
+
+    EXPECT(m1 == m2);
+}
+
+TEST_CASE(simplify_zero_mul_const_vec)
+{
+    migraphx::shape inner{migraphx::shape::int32_type, {2}};
+    migraphx::shape outer{migraphx::shape::int32_type, {1, 2, 3, 3}};
+    migraphx::op::broadcast b{1, {1, 2, 3, 3}};
+    migraphx::module m1;
+    {
+        auto zero    = m1.add_literal({inner, {0, 0}});
+        auto x       = m1.add_parameter("x", outer);
+        auto zerob   = m1.add_instruction(b, zero);
+        auto mul_ins = m1.add_instruction(migraphx::make_op("mul"), x, zerob);
+        m1.add_return({mul_ins});
+    }
+    run_pass(m1);
+
+    migraphx::module m2;
+    {
+        auto zero = m2.add_literal({inner, {0, 0}});
+        m2.add_parameter("x", outer);
+        auto zerob = m2.add_instruction(b, zero);
+        m2.add_return({zerob});
+    }
+
+    EXPECT(m1 == m2);
+}
+
+TEST_CASE(simplify_zero_mul_const_vec2)
+{
+    migraphx::shape inner{migraphx::shape::int32_type, {2}};
+    migraphx::shape outer{migraphx::shape::int32_type, {1, 2, 3, 3}};
+    migraphx::op::broadcast b{1, {1, 2, 3, 3}};
+    migraphx::module m1;
+    {
+        auto zero    = m1.add_literal({inner, {0, 0}});
+        auto x       = m1.add_parameter("x", outer);
+        auto zerob   = m1.add_instruction(b, zero);
+        auto mul_ins = m1.add_instruction(migraphx::make_op("mul"), zerob, x);
+        m1.add_return({mul_ins});
+    }
+    run_pass(m1);
+
+    migraphx::module m2;
+    {
+        auto zero = m2.add_literal({inner, {0, 0}});
+        m2.add_parameter("x", outer);
+        auto zerob = m2.add_instruction(b, zero);
+        m2.add_return({zerob});
+    }
+
+    EXPECT(m1 == m2);
+}
+
+TEST_CASE(simplify_zero_div_const)
+{
+    migraphx::module m1;
+    {
+        auto zero    = m1.add_literal(0);
+        auto x       = m1.add_parameter("x", {migraphx::shape::int32_type, {1}});
+        auto div_ins = m1.add_instruction(migraphx::make_op("div"), zero, x);
+        m1.add_return({div_ins});
+    }
+    run_pass(m1);
+
+    migraphx::module m2;
+    {
+        auto zero = m2.add_literal(0);
+        m2.add_parameter("x", {migraphx::shape::int32_type, {1}});
+        m2.add_return({zero});
+    }
+
+    EXPECT(m1 == m2);
+}
+
+TEST_CASE(simplify_zero_div_const_vec)
+{
+    migraphx::shape inner{migraphx::shape::int32_type, {2}};
+    migraphx::shape outer{migraphx::shape::int32_type, {1, 2, 3, 3}};
```

Comment:
instead of `inner` and `outer` can you use `x_shape` and `zero_shape`? So that it is easier to follow when reading.

Reply:
Yup I can do that.

Reply:
Done